src/neuroglancer/viewer.ts

```diff
@@ -617,50 +617,55 @@ export class Viewer extends RefCounted implements ViewerState {
     if (urlParams.has('json_url')) {
       let json_url = urlParams.get('json_url');
       history.replaceState(null, '', removeParameterFromUrl(window.location.href, 'json_url'));
-      sendHttpRequest(openHttpRequest(json_url!), 'json')
+      StatusMessage
+      .forPromise(
+        sendHttpRequest(openHttpRequest(json_url!), 'json')
           .then(response => {
             this.state.restoreState(response);
-          })
-          .catch(() => {
-            console.log('failed to load from: ' + json_url);
-          });
-    }
+          }),
+              {
+                initialMessage: `Retrieving state from json_url: ${json_url}.`,
+                delay: true,
```

Comment:
Oh, haven't used `forPromise` before. Looks like the `delay` parameter is the time until the message *appears*, not the time until it *disappears* - which makes the message now completely invisible to the user :(
The `showTemporaryMessage` function has the more useful `closeAfter` parameter. Sorry 'bout that.

Reply:
okay.. i think i've fixed this

Reply:
wait.. this one is good... it shows the error message when it errors and doesn't show much if the server responds quickly... which is what we want.